### PR TITLE
rename connection params host->hostname and change port type

### LINKED ISSUE #113: IConnectionParams port and host type and naming convention
IConnectionParams should be modelled like:

```
interface IConnectionParams {
    hostname?: string
    port?: number
}
```

Rather than the current

```
interface IConnectionParams {
    host?: string
    port?: string
}
```

I understand this is probably inherited  from `node-postgres` but there is also an opportunity to align simple things like this in Deno user land from the start. I believe this aligns better with the convention currently set by `Deno.listen` API. In addition, this is what is used in https://github.com/keroxp/deno-redis . I've also used it for https://github.com/lenkan/deno-amqp.

If you agree, I can submit a PR with the changes. If you don't agree, feel free to close this issue.

### PR DESCRIPTION
**Pull request by lenkan**
Closes #113 

I took the liberty to also refactor the creation of the connection_params.ts params because I found the code a bit hard to follow and modify. Let me know what you think.

I also think `parseDsn` should be moved into `connection_params.ts` and tested through createParams instead. If you agree, I'll do that as well.

### CODE REVIEW

File: connection_params.ts [hunk 1]
```diff
@@ -17,106 +18,117 @@ function getPgEnv(): IConnectionParams {
...
-  for (const source of sources) {
-    if (source[key]) {
-      return source[key];
-    }
-  }
-
-  return undefined;
+function isDefined<T>(value: T): value is Exclude<T, undefined | null> {
+  return value !== undefined && value !== null;
 }
```
Comment (bartlomieju) [inline connection_params.ts:23]: I think this could be simply:
```
function assert(val): asserts expr {}
```

File: connection_params.ts [hunk 2]
```diff
@@ -17,106 +18,117 @@ function getPgEnv(): IConnectionParams {
...
-export interface IConnectionParams {
+export interface ConnectionOptions {
   database?: string;
-  host?: string;
-  port?: string;
+  hostname?: string;
+  port?: number;
   user?: string;
   password?: string;
   application_name?: string;
```
Comment (bartlomieju) [inline connection_params.ts:38]: This field should be camelCased, could you do that @lenkan ?

File: connection_params.ts (hunk 1 above)
Comment (lenkan) [reply connection_params.ts]: Do you mean just invoking the built in `assert` function in place of this?

This type guard function is used in 3 places, and only one of them could actually be replaced with an assertion function, in the other two places it is just used as a type guard.

When I did `assert(val !== null && val !== undefined)` for the assertion, typescript did not recognise that `val` would be non-nullable after that line. So I lost the type safety.

I did however replace the `Exclude<T, undefined | null>` return type with the built in type `NonNullable<T>`.

File: connection_params.ts (hunk 2 above)
Comment (lenkan) [reply connection_params.ts]: :+1: I renamed this one to `applicationName`.

Should I also rename the internally used `ConnectionParams.application_name` to camelCase? 

Also, what about the dsn-style parameters?

```
postgres://server/dbname?application_name=appName
```

or 


```
postgres://server/dbname?applicationName=appName
```

File: connection_params.ts (hunk 1 above)
Comment (bartlomieju) [reply connection_params.ts]: > Do you mean just invoking the built in `assert` function in place of this?

There's no built-in `assert`. I guess this solution is fine as well.

File: connection_params.ts (hunk 2 above)
Comment (bartlomieju) [reply connection_params.ts]: DSN should behave as-is; ie. keep the snake_case, I'm fine with changing all internal usages to camelCase as well 👍

File: connection_params.ts (hunk 1 above)
Comment (lenkan) [reply connection_params.ts]: Yeah thanks, I don't know how I figured that. I had node typings loaded for some reason when I just tried it in the editor.

File: connection_params.ts (hunk 2 above)
Comment (lenkan) [reply connection_params.ts]: Great! Fixed it.
